ssl/statem/statem_clnt.c

```diff
@@ -3336,7 +3336,12 @@ int tls_construct_client_certificate(SSL *s, WPACKET *pkt)
                     SSL3_CC_HANDSHAKE | SSL3_CHANGE_CIPHER_CLIENT_WRITE))) {
         SSLerr(SSL_F_TLS_CONSTRUCT_CLIENT_CERTIFICATE,
                SSL_R_CANNOT_CHANGE_CIPHER);
-        goto err;
+        /*
+         * This is a fatal error, which leaves
+         * enc_write_ctx in an inconsistent state
+         * and thus ssl3_send_alert may crash.
+         */
+        return 0;
```

Comment:
Hmmm....

There are at least three problems with sending an alert here:
- one is that the SSL object does not know if there was a fatal error or just a protocol error.
- and the other is that sending the alert might block
- and returning SSL_ERROR_SSL instructs the application not to continue calling SSL functions

and therefore the incomplete alert wont get sent at all...